e2e/libs/node_exec/node_exec.py

```diff
@@ -1,39 +1,58 @@
+import time
+import logging
+
+from utils.common_utils import k8s_core_api
 from kubernetes import client
 from kubernetes.stream import stream
-
-import time
+from kubernetes.client.rest import ApiException
 
 DEFAULT_POD_TIMEOUT = 180
 DEFAULT_POD_INTERVAL = 1
 
+
 class NodeExec:
 
-    def __init__(self, namespace):
-        self.namespace = namespace
-        self.core_api = client.CoreV1Api()
+    def __init__(self, case_name):
+        self.namespace = case_name[0:62]
         self.node_exec_pod = {}
+        # maximum length of namespace is 63 characters
         namespace_manifest = {
             'apiVersion': 'v1',
             'kind': 'Namespace',
             'metadata': {
                 'name': self.namespace
             }
         }
-        self.core_api.create_namespace(
+
+
+        k8s_core_api().create_namespace(
             body=namespace_manifest
         )
 
     def cleanup(self):
+        logging.info("cleaning up node related resources")
+
         for pod in self.node_exec_pod.values():
-            res = self.core_api.delete_namespaced_pod(
-                name=pod.metadata.name,
-                namespace=self.namespace,
-                body=client.V1DeleteOptions()
+            try:
+                resp = k8s_core_api().delete_namespaced_pod(
```

Comment:
Why not use `self.core_api`?

Reply:
Yes, I've updated to get the `k8s_core_api` API instance in the initial function and operate with that API instance.
https://github.com/longhorn/longhorn-tests/blob/ccca467468a21e0181a592e0d238aefa1a9a3424/e2e/libs/node_exec/node_exec.py#L13-L16
https://github.com/longhorn/longhorn-tests/blob/ccca467468a21e0181a592e0d238aefa1a9a3424/e2e/libs/node_exec/node_exec.py#L132-L135